src/spikeinterface/extractors/mdaextractors.py

```diff
@@ -216,10 +216,14 @@ def write_sorting(sorting, save_path, write_primary_channels=False):
         times_list = []
         labels_list = []
         primary_channels_list = []
-        for unit_id in unit_ids:
+        for unit_id_i, unit_id in enumerate(unit_ids):
```

Comment:
use unit_index to be align with other modules

Reply:
Thanks Sam!

Reply:
If unit IDs are non-integer (which I've had), this would fail unless an index is created (type problem later on).

Reply:
Hang on, this would discard unit_ids if they are set in non-increasing manner. MDA requires integers, so if there are numbers we should use these (HDSort for example produces strange unit numbers, and I like to save stuff intio mda, but want to retain the original ids).

Reply:
Have you tried sorting.save(folder=...) 
? That's what we use and we recommend!

Reply:
this was just a naming issue